File: Proyecto/Windows phone/Messages/Messages/DB/BDHelp.cs
using System;
using System.Collections.Generic;
using System.Data.SQLite;
using System.Diagnostics;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Messages.DB
{
    class BDHelp
    {
        SQLiteConnection ConnectionVar;
        public BDHelp()
        {
            ConnectionVar = new SQLiteConnection("Data Source=Student.db");
            ConnectionVar.Open();
            var Command = ConnectionVar.CreateCommand();
            Command.CommandText = "CREATE TABLE IF NOT EXISTS messages (id_message INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, sender INTEGER NOT NULL, receiver INTEGER NOT NULL, message VARCHAR(50) NOT NULL, date_message VARCHAR(30));";
            Command.ExecuteNonQuery();
        }
        public void Insert(String Sender, String Receiver, String Message, String DateMessage)
        {
            var Command = ConnectionVar.CreateCommand();
            Command.CommandText = "INSERT INTO messages(sender, receiver, message, date_message) VALUES(" +Sender+", "+Receiver+ ", '" + Message + "', '" + DateMessage + "')";
            Command.ExecuteNonQuery();
        }
        public List<OneMessage> QueryMessages(String Sender, String User)
        {
            List<OneMessage> Messages = new List<OneMessage>();
            var Command = ConnectionVar.CreateCommand(); 
            Command.CommandText = "SELECT message, sender, id_message FROM messages WHERE(sender = " +Sender+" AND receiver = "+User+") OR(receiver = "+Sender+" AND sender = "+User+") ORDER BY id_message ASC";
            SQLiteDataReader sdr = Command.ExecuteReader();
            int Index = 0;
            while (sdr.Read())
            {
                Messages.Add(new OneMessage());
                Messages.ElementAt(Index).Sender= sdr.GetInt32(1);
                Messages.ElementAt(Index).Message = sdr.GetString(0);
                Debug.WriteLine(Messages.ElementAt(Index).Sender);
                Debug.WriteLine(Messages.ElementAt(Index).Message);
                Debug.WriteLine(sdr.GetInt32(2));
                Index += 1;
            }
            return Messages;
         }
        public void Close()
        {
            ConnectionVar.Close();
        }
            

    }
}

File: Proyecto/iOS/Messages/Messages/BDhelp.swift
//
//  BDhelp.swift
//  Messages
//
//  Created by Estudiantes on 4/06/16.
//  Copyright © 2016 Estudiantes. All rights reserved.
//

import Foundation
import SQLite

class BDhelp
{
    private var db:Connection
    private var messages: Table
    private var id_message: Expression<Int>
    private var sender: Expression<Int>
    private var receiver: Expression<Int>
    private var message: Expression<String>
    private var date_message: Expression<String>
    init()
    {
        let path = NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true).first!
        
        db = try! Connection("\(path)/db.sqlite3")
        print("\(path)")
        
        messages = Table("messages")
        id_message = Expression<Int>("id_message")
        sender = Expression<Int>("sender")
        receiver = Expression<Int>("receiver")
        message = Expression<String>("message")
        date_message = Expression<String>("date_message")
        
        try! db.run(messages.create(ifNotExists: true) { t in
            t.column(id_message, primaryKey: PrimaryKey.Autoincrement)
            t.column(sender)
            t.column(receiver)
            t.column(message)
            t.column(date_message)
            })
    }
    func insert(Sender: Int, Receiver: Int, Message: String, DateMessage: String)
    {
        try! db.run(messages.insert(sender <- Sender, receiver <- Receiver,message <- Message, date_message <- DateMessage))
    }
    func queryMessages(Sender: String) -> Array<OneMessage>//[OneMessage]
    {
        let query: String = "SELECT message,sender,id_message FROM messages WHERE (sender = "+Sender+" AND receiver = "+Connect.User+") OR (receiver = "+Sender+" AND sender = "+Connect.User+" ) ORDER BY id_message ASC"
        var messages:Array<OneMessage>=Array<OneMessage>()
        for sms in try! db.prepare(query){
            messages.append(OneMessage(send:String(sms[1]!),message:String(sms[0]!)))
        }
        return messages
    }
}